src/SonarScanner.MSBuild/BootstrapperClass.cs

```diff
@@ -166,6 +166,7 @@ private int PostProcess()
             {
                 var postProcessor = this.processorFactory.CreatePostProcessor();
                 succeeded = postProcessor.Execute(this.bootstrapSettings.ChildCmdLineArgs.ToArray(), config, teamBuildSettings);
+                CleanSonarQubeDirectory();
```

Comment:
This will try to delete everything inside `.sonarqube` directory. Like this, we'll lose possibility to debug and provide support to our customers because there will be no easy way to get real analysis properties file, internal state, protobuf files and UCFG files.

Let's find another way to do this. For example, I don't like `$(SonarQubeTempPath) != ''` to be the main decision condition. We should have something line `SonarRunAnalysis=='true'` that would be set after the `SonarQubeTempPath` is successfully computed. 

Having that, we can have very small `Sonar.Activation.targets` file that would be included in `SonarQube.Integration.targets` and the lifecycle would be something like:
* begin step: Produce `Sonar.Activation.targets` that would do nothing. Maybe just log that `SonarRunAnalysis` is true.
* end step: Modify `Sonar.Activation.targets` to set `SonarRunAnalysis` to `false` and log, that analysis is disabled, because it was already executed.

Like this, we would have simple (de)activation mechanism and possibility to inspect the `.sonarqube` directory. There might also be other way to solve it, this is just random idea from my limited knowledge of S4MSB :)

Reply:
I agree with not deleting the whole directory. Previously, the folder cleanup ws done in the `begin` step so the `out` and `conf` folders are available for inspection when diagnosing problems.

@pavel-mikula-sonarsource's suggestion should work. A simpler solution would be delete the targets file under the `.sonarqube\bin\targets\` folder and change the `SonarQubeImportBeforeInfo` target in the `ImportsBefore` targets not to raise an error (either remove the check completely, or write a message that the second targets file was not found).

Reply:
Ok, yes, that makes sense. I will not have time to work on it until the release, so i'll remove it from the 5.2 milestone. Thanks !